packages/pages-components/src/components/schema/yextFields/hours.tsx

```diff
@@ -1,9 +1,46 @@
 import { HoursType, DayType } from "../../hours/types.js";
 
+const validateHoursType = (hours: any): hours is HoursType => {
+  if (typeof hours !== "object") {
+    return false;
+  }
+  return (
+    "monday" in hours ||
+    "tuesday" in hours ||
+    "wednesday" in hours ||
+    "thursday" in hours ||
+    "friday" in hours ||
+    "saturday" in hours ||
+    "sunday" in hours
+  );
+};
+
+const validateDayType = (hours: any): hours is DayType => {
+  if (
+    typeof hours === "object" &&
+    "isClosed" in hours &&
+    Array.isArray(hours.openIntervals)
+  ) {
+    if (hours.openIntervals.length === 0) {
+      return true;
+    }
+    let result = true;
+    hours.openIntervals.forEach((interval: any) => {
+      result =
```

Comment:
Tiny optimization would be to return false as soon as result is false instead of going through the whole list, but doesn't really matter.